### ISSUE TITLE
"File Status Auto-Detection = Disable" ignored

### ISSUE THREAD

**Issue by itschytoo** [notepad-plus-plus/notepad-plus-plus#9815]
### Description of the Issue
When using notepad++ over sshfs every save triggers a modal dialog: "_This file has been modified by another program. Do you want to reload it?_"

The triggered modified event might be an error in the sshfs implementation, but npp creates the notification even with File Status Auto-Detection **disabled**.

If File Status Auto-Detection is enabled AND asked to Update silently the notification does not appear (but of course then the undo history is lost).

### Steps to Reproduce the Issue
1. Install WinFSP and sshfs-win from here: https://github.com/billziss-gh/sshfs-win
2. Mount a drive via sshfs
3. edit any file on said drive and save it

### Expected Behavior
When File Status Auto-Detection is disabled a file changed event should not trigger the modal dialog

### Actual Behavior
With disabled File Status Auto-Detection a modal dialog still appears every time

### Debug InformationNotepad++ v7.9.5   (64-bit)
Build time : Mar 21 2021 - 02:13:17
Path : D:\Programme\Notepad++\notepad++.exe
Admin mode : OFF
Local Conf mode : ON
OS Name : Windows 10 Pro (64-bit) 
OS Version : 2009
OS Build : 19042.928
Current ANSI codepage : 1252
Plugins : ComparePlugin.dll Linter.dll mimeTools.dll NppConverter.dll NppExec.dll NppExport.dll NppFTP.dll NPPJSONViewer.dll NppMarkdownPanel.dll XMLTools.dll 

The additional problem (can not attribute file change to own operation) seems to be an old one.
https://softwarerecs.stackexchange.com/questions/13875/windows-sshfs-sftp-mounting-clients
One comment (Michael Kohne Nov 20 '14 at 13:33)  states that npp does not correctly detect file changes over sshfs (here with a different sshfs-driver):

**Comment by mere-human** [comment]
I had an idea how to fix this.
So, we can add a checkbox to a dialog that says "Do not show this again: ☑".
If checked, the value a user has chosen, is remembered.
If the user wants to change their decision, there could be a section in the _Presences_ to reset the dialog check boxes.
This can also help to solve a similar user request about "Keep this file in editor" : https://community.notepad-plus-plus.org/topic/21092/keep-this-file-in-editor-annoying

**Comment by itschytoo** [comment]
That would be fine with me (I also occasionally suffer from the "keep this file in editor"-popup).

But what would distinguish this "disable the change notification from now on" feature from the existing "disable the change notification, period" feature?

I guess you propose to not change the underlying code but just suppress the popup?

As I said: fine with me.

**Comment by mere-human** [comment]
@itschytoo 

> I guess you propose to not change the underlying code but just suppress the popup?

Right. So it will always reload (or will not) depending on user decision.

---

Also, have you tried this _Monitoring_ option? It reloads the documents without any prompt.
![image](https://user-images.githubusercontent.com/9664141/116534337-10f1e000-a8eb-11eb-93d2-c6b8b4d4b700.png)

**Comment by itschytoo** [comment]
@mere-human 

The tail -f does not allow editing the file if active, so it does not serve.

The popup suppression feels kind of "wrong" but would make npp usable with sshfs.

**Comment by mere-human** [comment]
> I had an idea how to fix this.
> So, we can add a checkbox to a dialog that says "Do not show this again: ☑".
> If checked, the value a user has chosen, is remembered.
> If the user wants to change their decision, there could be a section in the _Presences_ to reset the dialog check boxes.
> This can also help to solve a similar user request about "Keep this file in editor" : https://community.notepad-plus-plus.org/topic/21092/keep-this-file-in-editor-annoying

@donho Please, take a look at this suggestion and provide your opinion, when you have time.
This could address a bunch of similar user requrests.

**Comment by itschytoo** [comment]
@donho 
That setting ist ignored (in connection with sshfs).
Please see my Initial comment.

**Comment by donho** [comment]
@itschytoo 
I do understand your issue and I will investigate it. It's due to sshfs surely.
But the comment above to @mere-human is regarding the issues of link he provided.
